### DISCUSSION TITLE
Extract a part of LineString enclosed by two coordinates

### DISCUSSION THREAD

**Discussion by mziadd** [NetTopologySuite/NetTopologySuite#631] [Q&A]
Using `ExtractLineByLocation` seems not working with the polygon ring, how i can make it work?

**Comment by FObermaier** [comment]
@mohammedX6, I don't know what you try to achieve. Could you reword your question, maybe illustrate it with some graphic?

**Comment by mziadd** [comment]
Sorry for misleading,

Suppose we have a polygon I want to the lines between Point F and Point C.

For Example, I want the lines

F->E->D->C
F->A->B->C

Many thanks

![180269012-690a796f-d1e8-47a2-b127-936940f545d8](https://user-images.githubusercontent.com/24648104/190433472-5db35398-a890-49ab-bc46-33d7d3b95506.jpg)

**Comment by FObermaier** [comment]
Have a look at this example:
https://github.com/NetTopologySuite/NetTopologySuite/blob/d605ea37a5c605298e38f40b1ad8351ee52de289/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs#L26-L59

**Comment by mziadd** [comment]
this will not work, maybe my clarification is not good, btw here is a better example : 
https://gis.stackexchange.com/questions/159758/polygon-points-to-polyline

**Reply by FObermaier** [reply]
You want a portion of the polygon's boundary?

**Reply by mziadd** [reply]
Yes that's what i want

**Comment by FObermaier** [comment]
This works for me:
```C#
// lineal geometry (geom), points (ptStart, ptEnd) from somewhere
var lip = new NetTopologySuite.LinearReferencing.LengthIndexOfPoint(geom);
var llm = new NetTopologySuite.LinearReferencing.LengthLocationMap(geom);
var llStart = llm.GetLocation(lip.IndexOf(ptStart.Coordinate));
var llEnd = llm.GetLocation(lip.IndexOf(ptEnd.Coordinate));

var result = NetTopologySuite.LinearReferencing.ExtractLineByLocation.Extract(geom, llStart, llEnd);
```

**Reply by mziadd** [reply]
It will not work if the start index is greater than the end index.

**Reply by FObermaier** [reply]
In that case you have to tie the two ends together yourself. Use `LineMerger`.

**Reply by mziadd** [reply]
I am a bit noob here, Can you please elaborate more, Thanks.
